added probe_scale parameter to define the viral inner compartment

fredericpoitevin opened the pull request:
the hard-coded value was too big for large viruses: it was set to half the object's Rg, which for large empty objects is smaller than the size of the void and resulted in failure. The `probe_scale` parameter can now be set by the user when the default value fails. For example, 0.1 works for RDV and PR772 but 0.5 does not, although it did for the smaller virus in examples/

fredericpoitevin committed:
added probe_scale parameter to define the viral inner compartment; the hard-coded value was too big for large viruses

chuckie82 commented on pysingfel/particle.py:
```diff
@@ -459,12 +459,12 @@ def create_masks(self):
         self.solvent_mask = self.solute_mask * ~self.create_solute_mask(dry=False)
 
-    def create_other_mask(self, virus_void=False):
+    def create_other_mask(self, virus_void=False, probe_scale=0.5):
```
I wonder whether probe_scale should be initialized as self.probe_scale in this method

chuckie82 replied:
Consider updating examples/notebooks/virus_filling.ipynb by explicitly passing probe_scale=0.5 in the example. 
So that people using the notebook is aware this is something that can be set

fredericpoitevin replied:
I will update the notebook soon after the numba/CUDA issue is resolved on psana.

chuckie82 replied:
I've merged this PR now. Let's update the notebook in a different PR

fredericpoitevin replied:
OK, will do that soon

fredericpoitevin replied:
@chuckie82 I took care of this in the "cupy" PR for Skopi

chuckie82 commented on pysingfel/particle.py:
```diff
@@ -459,12 +459,12 @@ def create_masks(self):
-    def create_other_mask(self, virus_void=False):
+    def create_other_mask(self, virus_void=False, probe_scale=0.5):
         """create_other_mask:
         Add another mask, True inside, False outside.
```
consider giving a more descriptive name (or more description in docstring here)

fredericpoitevin committed:
added self.other_mask_name=None and self.other_mask_probe_scale=0.5 to the Particle Class init method, as well as setters for them.

The create_masks function now creates self.other_mask in addition to 
self.solute_mask and self.solvent_mask.

self.other_mask is actually built by create_other_mask who does not take 
arguments anymore (it relies on the two defined in init now). At the 
moment only self.other_mask_name='virus_void' is implemented. It calls 
create_virus_void_mask() who will call ndimage.binary_fill_holes 
function to fill the virus void using a sphere structure of radius 
define as self.other_mask_probe_scale times the solute's Rg.

fredericpoitevin committed:
Merge branch 'master' of github.com:fredericpoitevin/pysingfel

fredericpoitevin committed:
resolved weird merged left

fredericpoitevin committed:
typo corrected in create_other_mask

fredericpoitevin committed:
Merge branch 'master' of https://github.com/chuckie82/pysingfel

fredericpoitevin committed:
Merge branch 'master' of https://github.com/chuckie82/pysingfel
